src/builtin.cpp

```diff
@@ -2923,6 +2925,17 @@ static int builtin_history(parser_t &parser, io_streams_t &streams, wchar_t **ar
                 show_time_format = w.woptarg ? w.woptarg : L"# %c%n";
                 break;
             }
+            case 'n': {
+                wchar_t *end = 0;
+                max_count = wcstol(w.woptarg, &end, 10);
+                if (!(*w.woptarg != L'\0' && *end == L'\0')) {
```

Comment:
Actually... in the specific case of getopt - is that at all needed if it has a required argument?